inflate: roll CRC calculation

If CONFIG_CRC32_TABLES is set, do tableless CRC calculation

 tiny-mpm/lib/inflate.c |   55 +++++++++++++++++++++++++++++++++----------------
 1 files changed, 38 insertions(+), 17 deletions(-)

Index: linux-2.6.16.19/lib/inflate.c
===================================================================
--- linux-2.6.16.19.orig/lib/inflate.c	2006-06-23 17:06:37.000000000 -0700
+++ linux-2.6.16.19/lib/inflate.c	2006-06-23 17:06:37.000000000 -0700
@@ -147,8 +147,9 @@
 static u8 window[0x8000]; /* use a statically allocated window */
 
 /* simple CRC calculation */
-static u32 crc_32_tab[256];
 #define CRCPOLY_LE 0xedb88320
+#ifdef CONFIG_CRC32_TABLES
+static u32 crc_32_tab[256];
 
 static void makecrc(void)
 {
@@ -169,6 +170,23 @@
 
 	return crc;
 }
+#else
+
+#define makecrc()
+
+static u32 crc32_le(u32 crc, unsigned char const *p, size_t len)
+{
+	int i;
+
+	while (len--) {
+		crc ^= *p++;
+		for (i = 0; i < 8; i++)
+			crc = (crc >> 1) ^ ((crc & 1) ? CRCPOLY_LE : 0);
+	}
+
+	return crc;
+}
+#endif /* crc tables */
 
 #else
 
